remote-trait-object/src/service/serde_support.rs

```diff
@@ -0,0 +1,91 @@
+// Copyright 2020 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use super::*;
+use serde::de::{Deserialize, Deserializer};
+use serde::ser::{Serialize, Serializer};
+
+pub struct SArc<T: ?Sized + Service> {
+    value: std::cell::Cell<Option<Arc<T>>>,
+}
+
+impl<T: ?Sized + Service> SArc<T> {
```

Comment:
This is weak a suggestion. I'm not strongly suggesting this.

How about splitting the SArc to `Exporter` and `Importer`?
`Exporter` only implements `Serialize`. `Importer` only implements `Deserialize`.

User will wrap their service object with Exporter when they want to export it.
User will use Importer<&dyn RemoteService> type to use a remote service object.

Reply:
@majecty Then what would appear on the actual signature of method in the service trait?
As I mentioned, the problem is that importer side and exporter side should observe a same interface.
Thus I think that is not possible.

Reply:
OK, I understood.